doc/stdenv/meta.chapter.md

```diff
@@ -45,6 +45,10 @@ Release branch. Used to specify that a package is not going to receive updates t
 
 The package’s homepage. Example: `https://www.gnu.org/software/hello/manual/`
 
+### `repository` {#var-meta-repository}
+
+HTTP-browsable source tree of the package. Automatically set if the package uses a `fetchFrom*` fetcher for its `src`. Example: `https://github.com/forthy42/gforth`
```

Comment:
Not every repository is HTTP(S)-browsable. 
Indeed [FTPs](http://ftp.freebsd.org/pub/FreeBSD/) were a very common thing some short time ago, and there are projects of hosting them on I2P networks.

Further, automatically filling this attribute is not a good idea. It should be explicitly set when desired.

```suggestion
Source repository of the package.

Examples:

- `https://git.savannah.gnu.org/git/gforth.git`
- `https://github.com/NixOS/nixops`
- `https://nest.pijul.com/pijul/pijul`
```

Reply:
> Not every repository is HTTP(S)-browsable.

no, but `meta.repository` is specifically for http-browsable source trees.  it exists for maintainer convenience and to be displayed by sites like search.nixos.org

Reply:
Why only HTTPS-browsable repositories? Why such discrimination against others?

Reply:
because this field is meant to be presented via a web browser.

would you accept a gemini or gopher link as a value for `meta.homepage`?

Reply:
Yes, I would.

Not a long time ago browsers displayed Gopher and even FTP. They dropped this because of code base cleanup.

What will be next? Dropping non-conformant forges like viewcvs?

Further, with some hack modern web browsers can rely on extensions that allow looking at those sites. 

Further, special purpose clients still exist for them.

Further, Emacs has support for them:

https://www.emacswiki.org/emacs/CategoryGopher/

https://www.emacswiki.org/emacs/AngeFtp/

Reply:
@AndersonTorres could you elaborate on this?

> Further, automatically filling this attribute is not a good idea. It should be explicitly set when desired.

I think this actually a great idea! When fetching from a repo, linking to that exact repo seems to be the sane thing to do, and exactly what I would expect as a user. Of course there are cases where that isn't possible (if you have multiple sources, when fetching from an unknown forge that we don't know how to translate, etc.), but those are the only ones that should be handled manually, IMO.

The other option of adding the new meta field to every single package manually results in additional (and unnecessary) maintainer load.